main.tf

```diff
@@ -64,25 +65,16 @@ resource "ibm_resource_instance" "cos_instance" {
 }
 
 locals {
-  cos_instance_id      = var.create_cos_instance == true ? tolist(ibm_resource_instance.cos_instance[*].id)[0] : tolist(data.ibm_resource_instance.cos_instance[*].id)[0]
-  cos_instance_guid    = var.create_cos_instance == true ? tolist(ibm_resource_instance.cos_instance[*].guid)[0] : tolist(data.ibm_resource_instance.cos_instance[*].guid)[0]
+  cos_instance_id      = var.create_cos_instance == true ? tolist(ibm_resource_instance.cos_instance[*].id)[0] : var.existing_cos_instance_id
+  cos_instance_name    = var.cos_instance_name == null ? "${var.environment_name}-cos" : var.cos_instance_name
```

Comment:
Can we update the variable for `var.cos_instance_name` to ensure it says if null is passed, and `var.create_cos_instance` is true, that the new COS instance name will default to `${var.environment_name}-cos`